src/sentry/data_export/tasks.py

```diff
@@ -113,20 +111,48 @@ def process_issues_by_tag(data_export, file, limit, environment_id):
             # See associated comment in './utils.py'
             gtv_list = convert_to_utf8(gtv_list_unicode)
             if limit and limit < next_offset:
-                # Since the next offset will pass the limit, write the remainder and quit
                 writer.writerows(gtv_list[: limit % SNUBA_MAX_RESULTS])
                 break
             else:
                 writer.writerows(gtv_list)
                 iteration += 1
 
 
-def process_discover(data_export, file):
-    # TODO(Leander): Implement processing for Discover
-    raise NotImplementedError("Discover processing has not been implemented yet")
+def process_discover(data_export, file, limit, environment_id):
+    """
+    Convert the discovery query to a CSV, writing it to the provided file.
+    """
+    try:
+        processor = DiscoverProcessor(
+            discover_query=data_export.query_info, organization_id=data_export.organization_id
+        )
+    except ExportError as error:
+        metrics.incr("dataexport.error", instance=six.text_type(error))
```

Comment:
This will be sampled by default so you might not get any results. Given that this should be low frequency you might want to set the sampling rate to 1.0.

Reply:
Done!